test/integration/document_test.go

```diff
@@ -129,15 +129,20 @@ func TestDocument(t *testing.T) {
 		go func() {
 			defer wg.Done()
 
-			// receive changed event.
-			resp := <-rch
-			if resp.Err == io.EOF {
-				return
-			}
-			assert.NoError(t, resp.Err)
+			for {
+				// receive changed event.
+				resp := <-rch
+				if resp.Err == io.EOF {
+					return
+				}
```

Comment:
```suggestion
				if resp.Err == io.EOF {
				        assert.Fail(t, resp.Err.Error())
					return
				}
```

Compare the contents of this test document at the end. So, This test is fully verifying the behavior we are currently testing.
How about adding something that triggers a test failure in case of an unintended response in this state? I think this will allow other developers to determine the cause of the failure more quickly. What do you think?